### PR TITLE
Support embedded videos in CMS pages

### PR DESCRIPTION
**Pull request by nikomakela**
KK-1001 KK-1006.

Installed a new version of the HCRC-lib. It also had a dependency for a later version of the HDS, so a new version of the Helsinki Design System was also needed.

The new version of the HCRC-lib needed some additions to the HCRC-configuration, so they are now added.

With the new versions and configurations the CMS pages now supports Youtube and Vimeo blocks for embedded videos.

image

----

These changes are needed from HCRC-lib: https://github.com/City-of-Helsinki/react-helsinki-headless-cms/pull/99

### CODE REVIEW

**Comment by nikomakela** [comment]
It seems like the hero image needs some kind of fix because it is now rendered empty

- [ ] Fix the cms page hero image (the missing fallback image)

**Comment by nikomakela** [comment]
> It seems like the hero image needs some kind of fix because it is now rendered empty
> 
> * [x]  Fix the cms page hero image (the missing fallback image)

image

File: src/common/components/formikWrappers/FormikDropdown.tsx [hunk 1]
```diff
@@ -12,6 +12,8 @@ type DropdownProps = Omit<
 
 type Props = DropdownProps & {
   name: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (HeikkM) [inline src/common/components/formikWrappers/FormikDropdown.tsx:15]: This is apparently needed?

File: src/common/components/formikWrappers/Combobox.tsx [hunk 2]
```diff
@@ -20,6 +20,8 @@ type DropdownProps = Omit<
 
 export type ComboboxProps = Omit<DropdownProps, 'toggleButtonAriaLabel'> & {
   name: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (HeikkM) [inline src/common/components/formikWrappers/Combobox.tsx:23]: This lintter bypass is apparently needed?

File: src/domain/profile/modal/__tests__/EditProfileModal.test.tsx [hunk 3]
```diff
@@ -88,23 +89,31 @@ it('renders snapshot correctly', () => {
...
+    }),
+    {
+      target: { value: formData.email },
+    }
+  );
+  fireEvent.change(getByRole('textbox', { name: 'Puhelinnumero *' }), {
     target: { value: formData.phoneNumber },
   });
-  fireEvent.change(getByLabelText('Etunimi *'), {
+  fireEvent.change(getByRole('textbox', { name: 'Etunimi *' }), {
```
Comment (HeikkM) [inline src/domain/profile/modal/__tests__/EditProfileModal.test.tsx:105]: Should we use testid for these to have more robust testing ids in english?
https://testing-library.com/docs/queries/bytestid/

**Review by HeikkM — APPROVED** [review]

File: src/domain/profile/modal/__tests__/EditProfileModal.test.tsx (hunk 3 above)
Comment (nikomakela) [reply src/domain/profile/modal/__tests__/EditProfileModal.test.tsx:105]: There is a good reason to use those roles: they tests the accessibility names so they tests how the screen reader reads the fields!

File: src/common/components/formikWrappers/Combobox.tsx (hunk 2 above)
Comment (nikomakela) [reply src/common/components/formikWrappers/Combobox.tsx:23]: Yep, when the `any` is used. I could try to change the `any` to some real type that the Combobox uses, but I tried for a while and didn't manage to get it to work. Didn't make much sense to fight against it.

File: src/domain/profile/modal/__tests__/EditProfileModal.test.tsx (hunk 3 above)
Comment (nikomakela) [reply src/domain/profile/modal/__tests__/EditProfileModal.test.tsx:105]: so, if you change the test language to English, then you need to use English texts too.

**Comment by nikomakela** [comment]
NOTE: This is still waiting for https://github.com/City-of-Helsinki/react-helsinki-headless-cms/pull/99 to be merged

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr466.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-ui-pr466.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**